packages/framework/ai-collab/src/explicit-strategy/agentEditTypes.ts

```diff
@@ -0,0 +1,177 @@
+ *
+ * TODO: Tests of range edits.
+ *
+ * TODO: Handle 429 rate limit error from OpenAI.
+ *
+ * TODO: Add an app-specific guidance string.
+ *
+ * TODO: Give the model a final chance to evaluate the result.
+ *
+ * TODO: Separate system prompt into [system, user, system] for security.
+ *
+ * TODO: Top level arrays are not supported with current DSL.
+ *
+ * TODO: Structured Output fails when multiple schema types have the same first field name (e.g. id: sf.identifier on multiple types).
+ *
+ * TODO: Pass descriptions from schema metadata to the generated TS types that we put in the prompt
+ */
+
+/**
+ * This is the field we force the LLM to generate to avoid any type ambiguity (e.g. a vector and a point both have x/y and are ambiguous without the LLM telling us which it means).
+ */
+export const typeField = "__fluid_type";
+
+/**
+ * A field that is  auto-generated and injected into nodes before passing data to the LLM to ensure the LLM can refer to nodes in a stable way.
+ */
+export const objectIdKey = "__fluid_objectId";
+
+/**
+ * Describes an edit to a field within a node.
+ * @remarks TODO: what is the [key: string] for?
+ */
+export interface TreeEditObject {
+	[key: string]: TreeEditValue;
+	[typeField]: string;
+}
+/**
+ * An array of {@link TreeEditValue}'s, allowing a single {@link TreeEdit} to contain edits to multiple fields.
+ */
+export type TreeEditArray = TreeEditValue[];
+
+/**
+ * The potential values for a given {@link TreeEdit}.
+ * @remarks These values are typically a field within a node or an entire node,
+ */
+export type TreeEditValue = JsonPrimitive | TreeEditObject | TreeEditArray;
+
+/**
+ * This is the the final object we expected from an LLM response.
+ * @remarks Because TreeEdit can be multiple different types (polymorphic),
+ * we need to wrap to avoid anyOf at the root level when generating the necessary JSON Schema.
+ */
+export interface EditWrapper {
+	// eslint-disable-next-line @rushstack/no-new-null
+	edit: TreeEdit | null;
+}
+
+/**
+ * Union type representing all possible types of edits that can be made to a tree.
+ */
+export type TreeEdit = Insert | Modify | Remove | Move;
+
+/**
+ * The base interface for all types of {@link TreeEdit}.
+ */
+export interface Edit {
+	explanation: string;
+	type: "insert" | "modify" | "remove" | "move";
+}
+
+/**
+ * This object provides a way to 'select' either a given node or a range of nodes in an array.
+ */
+export type Selection = ObjectTarget | Range;
+
+/**
+ * A Target object for an {@link TreeEdit}, identified by the target object's Id
+ */
+export interface ObjectTarget {
+	target: string;
+}
+
+/**
+ * Desribes where an object can be inserted into an array.
+ * For example, if an you have an array with 5 objects, and you insert an object at index 3, this differentiates whether you want
+ * the existing item at index 3 should be shifted forward (if the 'place' is 'before') or shifted backwards (if the 'place' is 'after')
```

Comment:
`place` doesn't seem to exist anymore? Something feels like it needs to be updated in these docs.

Reply:
Yup-- replaced the 'place' word with 'location' and tiny reword in commit 467a492cb2bcca4b3ead99d22f61af27cc063054